frontend/src/utilities/__tests__/time.spec.ts

```diff
@@ -1,6 +1,205 @@
+    expect(ensureTimeFormat('3:30PM')).toBe('3:30 PM');
+  });
+});
+
+describe('printSeconds', () => {
+  it('should print seconds', () => {
+    expect(printSeconds(3661)).toBe('1 second, 1 minute, 1 hour');
+  });
+
+  it('should handle a single unit', () => {
+    expect(printSeconds(5)).toBe('5 seconds');
+  });
+
+  it('should handle a single unit at max', () => {
+    expect(printSeconds(60)).toBe('1 minute');
+  });
+});
+
+describe('relativeTime', () => {
+  it('should convert milliseconds to minutes and seconds', () => {
+    expect(relativeTime(102, 30)).toBe('Just now');
+  });
+
+  it('should print ago if the current time is less than the previous', () => {
+    expect(relativeTime(62000, 1)).toBe('1 minute ago');
+  });
+
+  it('should print in if the elapsed time is less than 0', () => {
+    expect(relativeTime(1, 700000)).toBe('in 12 minutes');
+  });
+
+  it('should print in if the elapsed time is less than 0', () => {
+    expect(relativeTime(1, 62000)).toBe('in 1 minute');
+  });
+
+  it('should return just now when previous is NaN', () => {
+    expect(relativeTime(1000, NaN)).toBe('Just now');
+  });
+
+  it('should return just now if elapsed is less than 0', () => {
+    expect(relativeTime(1000, 2000)).toBe('Just now');
+  });
+
+  it('should return formatted time in minutes when elapsed time is less than an hour', () => {
+    const current = Date.now();
+    const previous = current - 30 * 60 * 1000; // 30 minutes ago
+    expect(relativeTime(current, previous)).toBe('30 minutes ago');
+  });
+
+  it('should return formatted time in hours when elapsed time is less than a day', () => {
+    const current = Date.now();
+    const previous = current - 6 * 60 * 60 * 1000; // 6 hours ago
+    expect(relativeTime(current, previous)).toBe('6 hours ago');
+  });
+
+  it('should return formatted time in days when elapsed time is less than a month', () => {
+    const current = Date.now();
+    const previous = current - 2 * 24 * 60 * 60 * 1000; // 2 days ago
+    expect(relativeTime(current, previous)).toBe('2 days ago');
+  });
+
+  it('should return formatted time in months when elapsed time is less than a year', () => {
+    const current = Date.now();
+    const previous = current - 2 * 30 * 24 * 60 * 60 * 1000; // 2 months ago
+    expect(relativeTime(current, previous)).toBe('2 months ago');
+  });
+
+  it('should return formatted date when elapsed time is more than a year', () => {
+    const current = Date.now();
+    const previous = current - 2 * 365 * 24 * 60 * 60 * 1000; // 2 years ago
+    expect(relativeTime(current, previous)).toMatch(/\d{1,2} [A-Za-z]+ \d{4}/); // Matches "01 Jan 2022" format
+  });
+
+  it('should get the correct month', () => {
+    const result = relativeTime(2500, 1610859600000);
+    expect(result).toBe('17 Jan 2021');
+  });
+
+  it('should get the correct month', () => {
+    const result = relativeTime(2500, 1613538000000);
+    expect(result).toBe('17 Feb 2021');
+  });
```

Comment:
I suggest that you combine all the month test cases into a single test case with multiple assertions. Tests don't always need to be so independent.
```suggestion
  it('should get the correct month', () => {
    expect(relativeTime(2500, 1610859600000)).toBe('17 Jan 2021');
    expect(relativeTime(2500, 1613538000000)).toBe('17 Feb 2021');
    ...
  });
```